Tests/Unit/EmogrifierTest.php

```diff
@@ -2121,4 +2118,41 @@ public function emogrifyMovesStyleElementFromHeadToBody()
             $result
         );
     }
+
+    private function htmlStringContainsTagWithAttribute($html, $tagName, $attribute)
+    {
+        return preg_match('/<' . preg_quote($tagName) . '[^>]+' . preg_quote($attribute) . '/', $html) > 0;
+    }
+
+    /**
+     * @test
+     */
+    public function emogrifyPrefersInlineStyleOverCssBlockStyleForHtmlAttributesMapping()
+    {
+        $this->subject->setHtml(
+            '<html><head><style>p {width:1px}</style></head><body><p style="width:2px"></p></body></html>'
+        );
+        $this->subject->enableCssToHtmlMapping();
+
+        $result = $this->subject->emogrify();
+
+        self::assertTrue($this->htmlStringContainsTagWithAttribute($result, 'p', 'width="2"'));
+    }
+
+    /**
+     * @test
+     */
+    public function emogrifyCorrectsHtmlAttributesMappingWhenMultipleMatchingRulesAndLastRuleIsAuto()
+    {
+        $this->subject->setHtml(
+            '<html><head><style>p {width:1px}</style></head><body><p class="autoWidth"></p></body></html>'
+        );
+        $this->subject->setCss('p.autoWidth {width:auto}');
+        $this->subject->enableCssToHtmlMapping();
+
+        $result = $this->subject->emogrify();
+
+        self::assertNotContains('width="1"', $result);
+        self::assertFalse($this->htmlStringContainsTagWithAttribute($result, 'p', 'width="1"'));
```

Comment:
This assertion is not needed anymore as it never can be true if the assertNotContains above is true.

Reply:
Of course, this happens when you are in a hurry...

On the other hand, I think checking just for the attribute name without a value (`width=`) would be better here, because we should make sure that the `width` attribute is missing, not just that it does not have the value `1`. What do you think?